### ISSUE TITLE
package dependencies

### ISSUE THREAD

**Issue by graemeblair** [DeclareDesign/DeclareDesign#299]
when you have a package dependency in a design and you just load the design, you can't print or draw_data or anything else. suggest we think of a way round this, like onload, that either prints what packages you need or even actually loads them. this could be as simple as detecting what packages are loaded when you do each declare step and concatenating them, and then `get_dependencies(design)`

otherwise hard to parse errors like:

```
> design
Joining, by = c("communities", "zones")
Error in block(blocking_data, n.tr = 2, groups = "zones", id.vars = "communities",  : 
  could not find function "block"
```

**Comment by macartan** [comment]
Makes a lot of sense to me: print and load if possible
If people have many packages loaded will these listed even if not needed
for the design?
Would dependencies get added as attributes to a design?

**Comment by nfultz** [comment]
I think each *step* should have it's own attribute of which packages were
used when it was declared.
